.github/workflows/test.yml

```diff
@@ -0,0 +1,34 @@
+name: CI
+
+on:
+  push:
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        emacs_version:
+          # Add more lines like this if you want to test on different Emacs versions.
+          - 27.2
+
+    steps:
+    - name: Set up Emacs
+      uses: purcell/setup-emacs@master
```

Comment:
I am unfamiliar with github workflows: this picks purcell's Github action to setup an Emacs version for testing.
Does it mean that if purcell deletes his repo this will break?

Reply:
Yes, but this would be a most unprecedented event. Nevertheless, if it does we can always switch to an alternative Emacs installation (there a few other choices around) or a fork of the removed repo.

Just to elaborate a little bit on how it works, this GitHub workflow sets up Continuous Integration (CI) on the GitHub servers. When a git push or pull event happens, GitHub allocates a clean Ubuntu server to run some *actions*, in our case we checkout the codebase, install Emacs using Purcell's action, and run `eldev test` on it. If the tests succeeds, then we get a green icon on GitHub when browsing the repository, otherwise a red cross indicating the latest push or pull request has broken something. The submitter then has to go back and fix the break. Thus, the integrity of the codebase is maintained across commits.

Back to our point, there are quite a few actions available on GitHub created by individuals and are used by thousands of GitHub projects, If the author deletes such a repository, then thousands of projects will fail CI. Although this is a possible scenario,it is most unlikely that someone will delete an action many depend on it without a good reason. These actions are really simple scripts that can be reproduce anytime or replaced by a fork someone has available.

Please also note this workflow sits on top of `Eldev`, we don't need to set up Continuous Integration. We can still use `eldev test` on our development machine, I thought it would have been nice to set up CI anyway. now that we have `Eldev` configured in the codebase :)

Reply:
Thanks for the detailed explanation! I was just curious to see if I understood things right. It all sounds great, thanks for introducing me to Eldev!!